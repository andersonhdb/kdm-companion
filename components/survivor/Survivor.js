import React from 'react'
import {
  View,
  Text,
  Title,
  Subtitle,
  Image,
  Icon,
  Button,
  Row,
  Caption,
  Divider,
} from '@shoutem/ui'

import { observer, inject } from 'mobx-react/native'
import PropTypes from 'prop-types'
import R from 'ramda'

import GenderButton from './GenderButton'
import EditableProperty from './EditableProperty'
import FightingArts from './FightingArts'
import Disorders from './Disorders'
import Note from '../Note'

const ico_movement = require('../../images/icon_movement-24.png')
const ico_accuracy = require('../../images/icon_accuracy-24.png')
const ico_strength = require('../../images/icon_strength-24.png')
const ico_evasion = require('../../images/icon_evasion-32.png')
const ico_luck = require('../../images/icon_luck-24.png')
const ico_d10 = require('../../images/icon_d10-32.png')
const ico_insanity = require('../../images/ico_insanity-32.png')

// This is the individual survivor screen
@inject(({ store }, props) => ({
  survivor: store.selectedCampaign.settlement.survivors.get(props.survivorId),
}))
@observer
export default class Survivor extends React.Component {
  static propTypes = {
    survivorId: PropTypes.string.isRequired, // ID of the survivor to show
  }

  render() {
    let survivor = this.props.survivor

    if (!survivor) {
      return null
    }

    return (
      <View>
        <View styleName="horizontal">
          <Text>Name: {survivor.name}</Text>
          <GenderButton
            gender={survivor.gender}
            changeGender={survivor.changeGender}
          />

          <EditableProperty
            label="Survival"
            help="(Limit: 3)"
            minimumValue={0}
            maximumValue={3}
            showLabel={true}
            quantity={survivor.survival}
            setQuantity={qty => survivor.setAttribute('survival', qty)}
          />
        </View>
        <View styleName="horizontal">
          <EditableProperty
            label="Movement"
            icon={ico_movement}
            quantity={survivor.movement}
            setQuantity={qty => survivor.setAttribute('movement', qty)}
          />
          <EditableProperty
            label="Accuracy"
            icon={ico_accuracy}
            quantity={survivor.accuracy}
            setQuantity={qty => survivor.setAttribute('accuracy', qty)}
          />
          <EditableProperty
            label="Strength"
            icon={ico_strength}
            quantity={survivor.strength}
            setQuantity={qty => survivor.setAttribute('strength', qty)}
          />

          <EditableProperty
            label="Evasion"
            icon={ico_evasion}
            quantity={survivor.evasion}
            setQuantity={qty => survivor.setAttribute('evasion', qty)}
          />
          <EditableProperty
            label="Luck"
            icon={ico_luck}
            quantity={survivor.luck}
            setQuantity={qty => survivor.setAttribute('luck', qty)}
          />
          <EditableProperty
            label="Speed"
            icon={ico_d10}
            quantity={survivor.speed}
            setQuantity={qty => survivor.setAttribute('speed', qty)}
          />
          <EditableProperty
            label="Insanity"
            icon={ico_insanity}
            quantity={survivor.insanity}
            setQuantity={qty => survivor.setAttribute('insanity', qty)}
          />
        </View>

        <Divider />

        <View>
          <Subtitle>Courage: 3</Subtitle>
          <View styleName="horizontal">
            <Text>Stalwart</Text>
            <Text>Stalwart2</Text>
            <Text>Stalwart3</Text>
          </View>
        </View>

        <Divider />

        <View>
          <Subtitle>Understanding: 3</Subtitle>
          <View styleName="horizontal">
            <Text>Analyze</Text>
            <Text>Analyze2</Text>
            <Text>Analyze3</Text>
          </View>
        </View>

        <Divider />

        <FightingArts
          fightingArts={survivor.fightingArts}
          addFA={survivor.addFA}
          removeFA={survivor.removeFA}
        />

        <Divider />

        <Disorders
          disorders={survivor.disorders}
          addDisorder={survivor.addDisorder}
          removeDisorder={survivor.removeDisorder}
        />

        <Divider />

        <Note
          title="Notes"
          notes={survivor.notes}
          saveNote={survivor.saveNotes}
        />
      </View>
    )
  }
}
