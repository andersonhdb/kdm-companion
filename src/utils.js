import R from 'ramda'
import { values } from 'mobx'

export const capitalize = str => str[0].toUpperCase() + str.slice(1)

export function uuid() {
  return ([1e7] + -1e3 + -4e3 + -8e3 + -1e11).replace(
    /[018]/g,
    a => (a ^ ((Math.random() * 16) >> (a / 4))).toString(16)[0]
  )
}

export function expansionFilter(map, expansion) {
  return R.filter(item => item.expansion.id === expansion.id, values(map))
}

/**
 *
 * @param {object} base object to merge data into (eg: getSnapshot(SettlementStat.create()))
 * @param {array} values array of objects with base attributes to merge (eg: values(self.innovations))
 * @param {array} attribute attribute from each values item to pick to merge into the base (eg: ['settlement', 'newborn'] )
 */
export function objectMerge(base, values, attribute) {
  let keys = Object.keys(base)
  let result = values.reduce((merged, item) => {
    if (R.path(attribute, item)) {
      // combine the 2 objects: current accumulated base and the one from the values item.
      return keys.reduce((keysAcc, key) => {
        return {
          ...keysAcc,
          [key]:
            typeof merged[key] === 'string'
              ? R.path(attribute, item)[key].length > 0
                ? merged[key] + R.path(attribute, item)[key] + '\n'
                : merged[key]
              : merged[key] + R.path(attribute, item)[key],
        }
      }, {})
    } else {
      return merged
    }
  }, base)

  return result
}
