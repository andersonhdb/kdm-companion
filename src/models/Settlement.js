import { types, getSnapshot, getParent } from 'mobx-state-tree'
import { SettlementBonus, init as defaultBonus } from './SettlementBonus'
import { Survivor, init as defaultSurvivor } from './Survivor'
import R from 'ramda'
import { values } from 'mobx'

// all are optional because it's used by innovations as well.
export const Settlement = types
  .model('Settlement', {
    name: 'New Settlement',
    survivalLimit: 0, // default is 0 because Settlement is also used for bonuses of innovations. This has to be setup to 1 for new Settlements.
    departing: types.optional(SettlementBonus, {}),
    newborn: types.optional(SettlementBonus, {}), // default survial = 1 needs to be setup for new settlements (same reason as above)
    showdown: types.optional(SettlementBonus, {}),
    survivors: types.map(Survivor),
    activeSurvivorsList: types.array(types.reference(Survivor)),
  })
  .actions(self => ({
    updateName(name) {
      self.name = name
    },
    add(settlement) {
      if (settlement.survivalLimit) {
        self.survivalLimit += settlement.survivalLimit
      }
      if (settlement.departing) {
        if (!self.departing) {
          self.departing = defaultBonus
        }
        self.departing.add(settlement.departing)
      }
      if (settlement.newborn) {
        if (!self.newborn) {
          self.newborn = defaultBonus
        }
        self.newborn.add(settlement.newborn)
      }
      if (settlement.showdown) {
        if (!self.showdown) {
          self.showdown = defaultBonus
        }
        self.showdown.add(settlement.showdown)
      }
    },
    remove(settlement) {
      // here a combination of destroy(item) + beforeDestroy hook on the item itself could be used to maybe simplify this stuff a little bit. does it work with references?
      if (settlement.survivalLimit) {
        self.survivalLimit -= settlement.survivalLimit
      }
      if (settlement.departing) {
        if (!self.departing) {
          self.departing = defaultBonus
        }
        self.departing.remove(settlement.departing)
      }
      if (settlement.newborn) {
        if (!self.newborn) {
          self.newborn = defaultBonus
        }
        self.newborn.remove(settlement.newborn)
      }
      if (settlement.showdown) {
        if (!self.showdown) {
          self.showdown = defaultBonus
        }
        self.showdown.remove(settlement.showdown)
      }
    },
    createSurvivor(name = undefined) {
      // TODO: this needs to be moved to the Survivor model!
      let survivorData = Object.assign(
        defaultSurvivor(),
        self.newborn ? self.newborn.bonus : {}
      )
      survivorData.survival = Math.min(
        survivorData.survival,
        self.survivalLimit
      )
      if (name) {
        survivorData.name = name
      }
      let survivor = Survivor.create(survivorData)
      survivor.applyNewbornMilestones() // TODO: this could be moved to afterCreate if that hook is only called when creating models and not when applying a snapshot.
      self.survivors.put(survivor)
      self.activeSurvivorsList.push(survivor)

      return survivor
    },
    handleSurvivorStatusChange(survivor) {
      if (survivor.status === 'dead') {
        self.deactivateSurvivor(survivor)
      } else {
        self.activateSurvivor(survivor)
      }
    },
    activateSurvivor(survivor) {
      if (!self.activeSurvivorsList.find(item => item.id === survivor.id)) {
        self.activeSurvivorsList.push(survivor.id)
      }
    },
    deactivateSurvivor(survivor) {
      let idx
      if (
        (idx = self.activeSurvivorsList.findIndex(
          item => item.id === survivor.id
        ))
      ) {
        self.activeSurvivorsList = self.activeSurvivorsList
          .slice(0, idx)
          .concat(self.activeSurvivorsList.slice(idx + 1))
      }
    },
    reorderSurvivor(survivor, position) {
      let idx = self.activeSurvivorsList.findIndex(
        item => item.id === survivor.id
      )
      if (idx >= 0) {
        // remove from array
        self.activeSurvivorsList = self.activeSurvivorsList
          .slice(0, idx)
          .concat(self.activeSurvivorsList.slice(idx + 1))
        // add in new position
        self.activeSurvivorsList.splice(position, 0, survivor.id)
      }
    },
  }))
  .views(self => ({
    filterSurvivors(status = 'alive') {
      if (status === 'alive') {
        return self.activeSurvivorsList
      } else {
        return R.filter(item => {
          return item.status === status
        }, values(self.survivors))
      }
    },
    get hasSOTF() {
      return getParent(self).hasSOTF
    },
  }))

export const init = {
  name: 'New Settlement',
  survivalLimit: 1,
  newborn: { survival: 1 },
}
