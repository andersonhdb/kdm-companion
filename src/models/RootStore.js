import { types, getSnapshot } from 'mobx-state-tree'
import { SettlementLocation } from './SettlementLocation'
import { Campaign } from './Campaign'
import { Innovation } from './Innovation'
import { Resource } from './Resource'
import { Monster } from './Monster'
import { Gear } from './Gear'
import { Expansion } from './Expansion'
import { Principle } from './Principle'
import { FightingArt, SecretFightingArt } from './FightingArt'
import { Disorder } from './Disorder'
import { Ability } from './Ability'
import { WeaponProficiency } from './WeaponProficiency'
import { Subscription } from './Subscription'
import { uuid } from '../utils'
import R from 'ramda'

import locationsData from '../data/settlement_locations'
import innovationsData from '../data/innovations'
import fightingArtsData from '../data/fightingArts'
import secretFightingArtsData from '../data/secretFightingArts'
import disordersData from '../data/disorders'
import resourceData from '../data/resources'
import monsterData from '../data/monsters'
import gearData from '../data/gear'
import expansionData from '../data/expansions'
import principlesData from '../data/principles'
import abilitiesData from '../data/abilities'
import weaponProficiencyData from '../data/weaponProficiencies'

export default types
  .model('RootStore', {
    locations: types.optional(types.map(SettlementLocation), locationsData),
    innovations: types.optional(types.map(Innovation), innovationsData),
    fightingArts: types.optional(types.map(FightingArt), fightingArtsData),
    secretFightingArts: types.optional(
      types.map(SecretFightingArt),
      secretFightingArtsData
    ),
    disorders: types.optional(types.map(Disorder), disordersData),
    resources: types.optional(types.map(Resource), resourceData),
    monsters: types.optional(types.map(Monster), monsterData),
    gear: types.optional(types.map(Gear), gearData),
    expansions: types.optional(types.map(Expansion), expansionData),
    principles: types.optional(types.map(Principle), principlesData),
    abilities: types.optional(types.map(Ability), abilitiesData),
    weaponProficiencies: types.optional(
      types.map(WeaponProficiency),
      weaponProficiencyData
    ),
    // stuff stored to disc after this
    version: 1,
    campaigns: types.optional(types.array(Campaign), [
      {
        id: 'new',
      },
    ]),
    selectedCampaign: types.optional(types.reference(Campaign), 'new'),
    subscription: types.optional(Subscription, {}),
  })
  .actions(self => ({
    createCampaign(name) {
      id = uuid()
      self.campaigns.push({ id, settlement: { name: name, survivalLimit: 1 } })
      self.selectCampaign(id)
    },
    selectCampaign(id) {
      self.selectedCampaign = id
    },
    deleteCampaign(id) {
      self.campaigns = R.reject(c => c.id === id, self.campaigns)
      if (self.campaigns.length === 0) {
        self.createCampaign('New Settlement')
      } else {
        self.selectedCampaign = self.campaigns[0]
      }
    },
    setVersion(version) {
      self.version = version
    },
    load(data) {
      self.campaigns = data.campaigns
      self.selectedCampaign = data.selectedCampaign
      self.version = data.version || 1
      self.subscription = data.subscription || {}
    },
  }))
  .views(self => ({
    get data() {
      return {
        campaigns: getSnapshot(self.campaigns),
        selectedCampaign: self.selectedCampaign.id,
        version: self.version,
        subscription: getSnapshot(self.subscription),
      }
    },
    get campaignExpansions() {
      return self.selectedCampaign.expansionList
    },
    selectedExpansionFilter(map) {
      return self.selectedCampaign.selectedExpansionFilter(map)
    },
    get availableLocations() {
      return R.sortBy(
        R.prop('name'),
        self.selectedExpansionFilter(self.locations)
      )
    },
    get availableInnovations() {
      return R.sortBy(
        R.prop('name'),
        self.selectedExpansionFilter(self.innovations)
      )
    },
    get availableFightingArts() {
      return R.sortBy(
        R.prop('name'),
        self.selectedExpansionFilter(self.fightingArts)
      )
    },
    get availableSecretFightingArts() {
      return R.sortBy(
        R.prop('name'),
        self.selectedExpansionFilter(self.secretFightingArts)
      )
    },
    get availableDisorders() {
      return R.sortBy(
        R.prop('name'),
        self.selectedExpansionFilter(self.disorders)
      )
    },
    get availableAbilities() {
      return R.sortBy(
        R.prop('name'),
        self.selectedExpansionFilter(self.abilities)
      )
    },
    get availableHunts() {
      return self.selectedExpansionFilter(self.monsters)
    },
    get availableWeaponProficiencies() {
      return self.selectedExpansionFilter(self.weaponProficiencies)
    },
    get numCampaigns() {
      return self.campaigns.length
    },
  }))
