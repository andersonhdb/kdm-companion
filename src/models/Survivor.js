import { types } from 'mobx-state-tree'
import { FightingArt } from './FightingArt'
import { Disorder } from './Disorder'
import { uuid } from '../utils'
import R from 'ramda'

const Survivor = types
  .model('Survivor', {
    id: types.identifier,
    name: types.string,
    fightingArts: types.array(types.reference(FightingArt)),
    disorders: types.array(types.reference(Disorder)),
    gender: types.optional(types.enumeration(['male', 'female']), 'male'),
    survival: 1,
    movement: 5,
    accuracy: 0,
    strength: 0,
    evasion: 0,
    luck: 0,
    speed: 0,
    insanity: 0,
    notes: '',
  })
  .actions(self => ({
    addFA(fa) {
      let found = R.find(item => fa.id === item.id, self.fightingArts)
      if (!found) {
        self.fightingArts.push(fa)
      }
    },
    removeFA(fa) {
      // TODO: removing an element other than the last gives an error.
      self.fightingArts.remove(fa)
    },
    addDisorder(disorder) {
      let found = R.find(item => disorder.id === item.id, self.disorders)
      if (!found) {
        self.disorders.push(disorder)
      }
    },
    removeDisorder(disorder) {
      // TODO: removing an element other than the last gives an error.
      self.disorders.remove(disorder)
    },
    changeGender() {
      let gender = self.gender === 'male' ? 'female' : 'male'
      self.gender = gender
    },
    setAttribute(attribute, quantity) {
      if (self[attribute] !== undefined) {
        self[attribute] = quantity
      }
    },
    saveNotes(notes) {
      self.notes = notes
    },
  }))
const init = () => ({ id: uuid(), name: 'Unnamed', fightingArts: [] })

export { Survivor, init }
