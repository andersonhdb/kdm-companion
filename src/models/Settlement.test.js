import { getSnapshot } from 'mobx-state-tree'
import { Settlement, init as newSettlementData } from './Settlement'
import { init as defaultSurvivor } from './Survivor'
import RootStore from './RootStore'

import * as utils from '../utils'

afterEach(() => {
  jest.restoreAllMocks()
})

describe('.create', () => {
  it('creates a barebones settlement', () => {
    let settlement = Settlement.create(newSettlementData)

    expect(getSnapshot(settlement)).toMatchSnapshot()
  })

  it('works with empty data', () => {
    let settlement = Settlement.create({})

    expect(getSnapshot(settlement)).toMatchSnapshot()
  })
})

describe('.add', () => {
  it('adds bonuses to settlement', () => {
    let settlement = Settlement.create(newSettlementData)
    let bonus = {
      survivalLimit: 1,
      departing: {
        survival: 3,
        insanity: 3,
        description: ['a description'],
      },
      newborn: {
        accuracy: 1,
        strength: 1,
        evasion: 1,
      },
      showdown: {
        description: ['another description'],
      },
    }
    settlement.add(bonus)
    expect(getSnapshot(settlement)).toMatchSnapshot()
  })
})

describe('.remove', () => {
  it('removes bonuses from settlement', () => {
    let settlement = Settlement.create(newSettlementData)
    let bonus = {
      survivalLimit: 1,
      departing: {
        survival: 3,
        insanity: 3,
        description: ['a description'],
      },
      newborn: {
        accuracy: 1,
        strength: 1,
        evasion: 1,
      },
      showdown: {
        description: ['another description'],
      },
    }
    settlement.add(bonus)
    settlement.remove(bonus)

    expect(getSnapshot(settlement)).toMatchSnapshot()
  })
})

describe('.createSurvivor', () => {
  it('creates a survivor and assings it to the settlement', () => {
    let settlement = Settlement.create(newSettlementData)

    let uuidMock = jest.spyOn(utils, 'uuid')
    uuidMock.mockImplementation(() => '4e55b990-5b16-480f-81d2-06765c52ec72')

    settlement.createSurvivor('test name')

    expect(getSnapshot(settlement)).toMatchSnapshot()
  })
})

describe('.filterSurvivors', () => {
  it('filters survivors by status', () => {
    let survivor1 = defaultSurvivor()
    let survivor2 = Object.assign(defaultSurvivor(), { status: 'dead' })

    let settlement = Settlement.create(
      Object.assign(newSettlementData, {
        survivors: {
          [survivor1.id]: survivor1,
          [survivor2.id]: survivor2,
        },
      })
    )

    let survivorList = settlement.filterSurvivors('alive')
    let deadList = settlement.filterSurvivors('dead')

    expect(survivorList).toHaveLength(1)
    expect(survivorList[0].id).toBe(survivor1.id)

    expect(deadList).toHaveLength(1)
    expect(deadList[0].id).toBe(survivor2.id)
  })
})

it('.hasSOTF throws if not attached', () => {
  let settlement = Settlement.create({})
  expect(() => settlement.hasSOTF).toThrow()
})

describe('with RootStore', () => {
  let store
  beforeEach(() => {
    store = RootStore.create()
  })

  describe('.hasSOTF', () => {
    it('has not sotf', () => {
      expect(store.selectedCampaign.settlement.hasSOTF).toBeFalsy()
    })

    it('has sotf', () => {
      store.selectedCampaign.selectPrinciple('newlife', { id: 'sotf' })
      expect(store.selectedCampaign.settlement.hasSOTF).toBeTruthy()
    })
  })
})
