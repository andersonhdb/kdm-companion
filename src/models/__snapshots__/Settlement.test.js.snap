// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.add adds bonuses to settlement 1`] = `
Object {
  "activeSurvivorsList": Array [],
  "departing": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [
      "a description",
    ],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 3,
    "strength": 0,
    "survival": 3,
    "understanding": 0,
  },
  "name": "New Settlement",
  "newborn": Object {
    "accuracy": 1,
    "courage": 0,
    "description": Array [],
    "evasion": 1,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 1,
    "survival": 1,
    "understanding": 0,
  },
  "showdown": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [
      "another description",
    ],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "survivalLimit": 2,
  "survivors": Object {},
}
`;

exports[`.create creates a barebones settlement 1`] = `
Object {
  "activeSurvivorsList": Array [],
  "departing": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "name": "New Settlement",
  "newborn": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 1,
    "understanding": 0,
  },
  "showdown": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "survivalLimit": 1,
  "survivors": Object {},
}
`;

exports[`.create works with empty data 1`] = `
Object {
  "activeSurvivorsList": Array [],
  "departing": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "name": "New Settlement",
  "newborn": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "showdown": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "survivalLimit": 0,
  "survivors": Object {},
}
`;

exports[`.createSurvivor creates a survivor and assings it to the settlement 1`] = `
Object {
  "activeSurvivorsList": Array [
    "4e55b990-5b16-480f-81d2-06765c52ec72",
  ],
  "departing": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "name": "New Settlement",
  "newborn": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 1,
    "understanding": 0,
  },
  "showdown": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "survivalLimit": 1,
  "survivors": Object {
    "4e55b990-5b16-480f-81d2-06765c52ec72": Object {
      "abilities": Array [],
      "accuracy": 0,
      "cannotUseAbilities": false,
      "cannotUseFightingArts": false,
      "cannotUseSurvival": false,
      "courage": 0,
      "disorders": Array [],
      "dragonTraits": Object {
        "accuracy": false,
        "championsRite": false,
        "courage": false,
        "destined": false,
        "fatedBlow": false,
        "frozenStar": false,
        "iridescentHide": false,
        "noble": false,
        "oraclesEye": false,
        "pristine": false,
        "reincarnated": false,
        "scar": false,
        "strength": false,
        "unbreakable": false,
        "understanding": false,
        "weaponMaster": false,
      },
      "evasion": 0,
      "fightingArts": Array [],
      "gender": "male",
      "hunt xp": 0,
      "id": "4e55b990-5b16-480f-81d2-06765c52ec72",
      "insanity": 0,
      "luck": 0,
      "movement": 5,
      "name": "test name",
      "notes": "",
      "rerollUsed": false,
      "skipNextHunt": false,
      "speed": 0,
      "status": "alive",
      "strength": 0,
      "survival": 1,
      "understanding": 0,
      "weaponProficiency": undefined,
      "weaponProficiencyLevel": 0,
    },
  },
}
`;

exports[`.remove removes bonuses from settlement 1`] = `
Object {
  "activeSurvivorsList": Array [],
  "departing": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "name": "New Settlement",
  "newborn": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 1,
    "understanding": 0,
  },
  "showdown": Object {
    "accuracy": 0,
    "courage": 0,
    "description": Array [],
    "evasion": 0,
    "hunt xp": 0,
    "insanity": 0,
    "strength": 0,
    "survival": 0,
    "understanding": 0,
  },
  "survivalLimit": 1,
  "survivors": Object {},
}
`;
